examples/remote_execution/buildbarn/README.md

```diff
@@ -1,14 +1,46 @@
-## Remote execution integration with Buildbarn
+# Remote execution integration with Buildbarn
 
 This project provides a small example of what a project that utilizes
 [Buildbarn](https://github.com/buildbarn).
 
 In this document, we will go over the key configs used in this setup. Using a
 local docker-compose deployment from the
-[example deployment repo](https://github.com/buildbarn/bb-deployments). If you
-already have a Buildbarn deployment you can skip that.
+[example deployment repo](https://github.com/buildbarn/bb-deployments).
+If you already have a `Buildbarn` deployment you can use that instead.
 
-### Deploy a local Buildbarn
+## Deploy a local Buildbarn
+
+### Set up a basic PATH
+
+`Buildbarn` runs all actions in a completely bare environment by default, and
+relies on the build client to specify the `$PATH` and other environment variables
+in the REAPI action message. As `buck2` does not send a default `$PATH` build
+will fail with errors about missing executables.
+
+Add the environment variables to the worker config you want
+`docker-compose/config/worker-fuse-ubuntu22-04.jsonnet` or
+`docker-compose/config/worker-hardlinking-ubuntu22-04.jsonnet`
+`.buildDirectories[].runners[]`, after the `workerId:` dictionary.
```

Comment:
I'm having some trouble parsing this sentence, did a part of it get cut out?

Reply:
Good point, thanks! I will revise it.

What I try to say (concisely) is: update whichever of the two paths you use, as you can choose either hardlinking or FUSE. Which for the sake of the example is problably bad. Hardlinking is the simplest but a little slower, we could clarify the example by dropping FUSE. Next, to update the proto file, which is a simple insertion but in a big structure, so here is a `jq`-style path for where to add it. Unfortunately this cannot be done with `yq` or `jq`.
And then, there is the big diff listing because it is still annoying to describe what to do :S.

I hope this is better:
```
Add the environment variables to the worker configs: `worker-fuse-ubuntu22-04.jsonnet` and `worker-hardlinking-ubuntu22-04.jsonnet`. The proto configuration specifies a `buildDirectories` list, with a `runner` list inside. There is only one runner in each config file. That runner should be modified to have the environment variables values:

      environment_variables: {
         PATH: '/bin:/usr/bin:/usr/local/bin',
      },

They can be put anywhere in the `runner`. The example diff below adds them at the end, after `workerId`.

It is also possible to make a crude insertion with `sed`:

     sed -i '/workerId/i \      environment_variables: { PATH: \'/bin:/usr/bin:/usr/local/bin\'},' docker-compose/config/worker-*-ubuntu22-04.jsonnet

```

Reply:
I think your original explanation was fine in principle, I just didn't quite understand the grammatical structure. Something like this would clear it up for me if it's accurate

> Add the environment variables to the worker config you want - normally either `path1` or `path2` - after the `workerId` field like this:

But the more detailed one is just as good, completely up to you